src/frontend/src/components/SearchPage/SearchPage.jsx

```diff
@@ -62,38 +67,74 @@ const SearchPage = () => {
             },
           };
         });
-      setResults(finalResults);
+      setResults({ type: 'author', searchResults: finalResults });
     },
   });
 
   // Hook that will re-render the page only if the state of searchText changes
   useEffect(() => {
-    executeSearch();
-  }, [searchText]);
+    async function searchPosts() {
+      const res = await fetch(`${telescopeUrl}/query?search=${searchText}`);
+      const searchResults = await res.json();
+      const postIds = searchResults.values.map((result) => result.id);
+      let posts = await Promise.all(postIds.map((id) => fetch(`${telescopeUrl}/posts/${id}`)));
```

Comment:
combine these two lines (79 and 80), you only need to walk this array once to get the id and look up the result.  Do we not return the URL to use in our JSON results already?

Reply:
Also, error handling on the fetch and json calls.

Reply:
so something like 
```
let postsInfo = await Promise.all(postIds.map(async (id) => {
  let post = await fetch(`https://dev.telescope.cdot.systems/posts/${id}`)
  return post.json();
}
```